src/async/pitfalls/pin.md

```diff
@@ -1,13 +1,20 @@
 # `Pin`
 
-When you await a future, all local variables (that would ordinarily be stored on
-a stack frame) are instead stored in the Future for the current async block. If
-your future has pointers to data on the stack, those pointers might get
-invalidated. This is unsafe.
-
-Therefore, you must guarantee that the addresses your future points to don't
-change. That is why we need to "pin" futures. Using the same future repeatedly
-in a `select!` often leads to issues with pinned values.
+All async blocks and async functions return types implementing the `Future`
+trait. When the async code awaits another future, the `Future` trait holds the
+local variables that would normally be located on the stack.
```

Comment:
Perhaps this would be a more direct way of saying what I suggested above:

```suggestion
Async blocks and functions return types implementing the `Future` trait.
The type returned is the result of a compiler transformation which turns
local variables into data stored in an enum.
```